app/src/main/java/com/spascoding/taskycourse/feature_agenda/presentation/detail_screen/DetailViewModel.kt

```diff
@@ -19,16 +21,31 @@ class DetailViewModel @Inject constructor() : ViewModel() {
 
             }
             DetailEvent.EditAction -> {
-                state.update {
-                    state.value.copy(isEditMode = true)
-                }
+                _state.value = state.value.copy(isEditMode = true)
+
             }
             DetailEvent.SaveAction -> {
-                state.update {
-                    state.value.copy(isEditMode = false)
-                }
+                _state.value = state.value.copy(isEditMode = false)
             }
             DetailEvent.PopBackStack -> {}
+            DetailEvent.EditTitleClick -> {}
+            DetailEvent.EditDescriptionClick -> {}
+        }
+    }
+
+    fun setNavigation(navigation: Navigation) = when (navigation) {
+        Navigation.EventDetailNavigation -> {
+            _state.value = state.value.copy(title = R.string.new_event)
+            _state.value = state.value.copy(description = R.string.event_description)
+        }
+        Navigation.RemainderDetailNavigation -> {
+            _state.value = state.value.copy(title = R.string.new_reminder)
+            _state.value = state.value.copy(description = R.string.reminder_description)
+        }
+        Navigation.TaskDetailNavigation -> {
+            _state.value = state.value.copy(title = R.string.new_task)
+            _state.value = state.value.copy(description = R.string.task_description)
         }
+        else -> {}
```

Comment:
These strings never change once you set them. If something never changes, it doesn't need to be state and therefore also doesn't need to be done by the ViewModel. You could put this logic in the UI instead and also get rid of your previous `setNavigation` call I commented on

Reply:
I will use this fields to hold event/reminder/task title and description.

Reply:
Yes, then they do change indeed. But if you need to do something once per screen (like setting these fields to their initial values), that should be done in the ViewModel's init block and not initiated from the UI. Doing this from the UI will reset the states after every configuration change like a screen rotation. 
You should have the information here which specific agenda item type the screen is for, so can't you process this logic in the initial state here in the ViewModel? You can access navigation parameters with `SavedStateHandle`